File: app/src/main/java/com/example/thebook/fragments/BodyFragment.java
package com.example.thebook.fragments;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebSettings;
import android.webkit.WebView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.thebook.MainActivity;
import com.example.thebook.R;

import io.reactivex.SingleObserver;
import io.reactivex.disposables.Disposable;


public class BodyFragment extends Fragment {

    private static final String TAG = "BodyFragment";

    private MainActivity activity;
    private WebView webView;
    private String baseUri = "file:///android_asset/";
    private String extension = ".html";

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.body_fragment, container, false);
        activity = (MainActivity) getActivity();

        webView = view.findViewById(R.id.webView);
        webView.setBackgroundColor(0x00000000);

        if (activity.singlePreferences != null) {
            activity.singlePreferences.subscribe(observerPref);
        }
        return view;
    }

    private SingleObserver<String> observerPref = new SingleObserver<String>() {
        @Override
        public void onSubscribe(Disposable d) {

        }
        @Override
        public void onSuccess(String s) {
            setWebView(baseUri + s + extension);
        }
        @Override
        public void onError(Throwable e) {
            e.printStackTrace();
        }
    };

    @SuppressLint("SetJavaScriptEnabled")
    private void setWebView(String uri) {
        webView.loadUrl(uri);
        WebSettings webSettings = webView.getSettings();
        webSettings.setBuiltInZoomControls(true);
        webSettings.setDisplayZoomControls(false);
        webSettings.setJavaScriptEnabled(true);
    }
}

File: app/src/main/java/com/example/thebook/fragments/BodyFragment.kt
package com.example.thebook.fragments

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.webkit.WebView
import androidx.fragment.app.Fragment

import com.example.thebook.MainActivity
import com.example.thebook.R

import io.reactivex.SingleObserver
import io.reactivex.disposables.Disposable


class BodyFragment : Fragment() {

    private val TAG = "BodyFragment"

    private var activity: MainActivity? = null
    private var webView: WebView? = null
    private val baseUri = "file:///android_asset/"
    private val extension = ".html"

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.body_fragment, container, false)
        activity = getActivity() as MainActivity?

        webView = view.findViewById(R.id.webView)
        webView!!.setBackgroundColor(0x00000000)

        if (activity!!.singleTitle != null) {
            activity!!.singleTitle?.subscribe(observerPref)
        }
        return view
    }

    private val observerPref = object : SingleObserver<String> {
        override fun onSubscribe(d: Disposable) {

        }
        override fun onSuccess(s: String) {
            setWebView(baseUri + s + extension)
        }
        override fun onError(e: Throwable) {
            e.printStackTrace()
        }
    }

    @SuppressLint("SetJavaScriptEnabled")
    private fun setWebView(uri: String) {
        webView!!.loadUrl(uri)
        val webSettings = webView!!.settings
        webSettings.builtInZoomControls = true
        webSettings.displayZoomControls = false
        webSettings.javaScriptEnabled = true
    }
}
